pkg/tests/ossm/smcp_annotation_test.go

```diff
@@ -86,6 +86,11 @@ func GetPodAnnotations(t TestHelper, podLocator oc.PodLocatorFunc) map[string]st
 		if err != nil {
 			t.Fatalf("Error parsing pod annotations json: %v", err)
 		}
+		if len(annotations) == 0 {
+			oc.DeletePod(t, podLocator)
+			oc.WaitPodReady(t, podLocator)
+			t.Fatal("Pod annotations are empty")
+		}
```

Comment:
This isn't okay. Now we retry only if there are no annotations, but we should retry if the assertions fail.

Reply:
Done, I moved the retry to outside the func to cover also assertions